addons/jobs/jobs-service/src/main/java/org/kie/kogito/jobs/service/repository/infinispan/InfinispanJobRepository.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.jobs.service.repository.infinispan;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.quarkus.infinispan.client.Remote;
+import io.vertx.core.Vertx;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.query.dsl.QueryFactory;
+import org.kie.kogito.jobs.service.model.JobStatus;
+import org.kie.kogito.jobs.service.model.ScheduledJob;
+import org.kie.kogito.jobs.service.qualifier.Repository;
+import org.kie.kogito.jobs.service.repository.ReactiveJobRepository;
+import org.kie.kogito.jobs.service.repository.impl.BaseReactiveJobRepository;
+
+import static org.kie.kogito.jobs.service.repository.infinispan.InfinispanConfiguration.Caches.SCHEDULED_JOBS;
+
+@ApplicationScoped
+@Repository(persistence = true)
```

Comment:
Do we really need this qualifier? Is this to allow to have in memory and infinispan in the same app?

Reply:
The idea here was to have multiple implementations of the interface, for instance one for in-memory and other for infinispan, in this way injecting the interface on beans cause exceptions, so the solution was to have only one `@Default` impl that delegates to the concrete class according to the config property for persistence. So I added the qualifier to solve this problem, do you have a better/simpler idea? I'll be happy to change, this was my last change to solve the problem to disable or enable persistence. 
One observation, since the injection is Lazy, the infinispan client is not being initialized by quarkus that why I'm injecting the `Instance<ReactiveJobRepository>` when persistence is not enabled, it evicts to get Exception during the startup.

Reply:
how about using named injections and then let people configure in application property which one to use? It will be more explicit and we could easily add more implementations. While the default would be in memory and we only look up beans when the config property is given. That way we could move all the code related to bootstrapping into the given bean that will be initialised only when that property is set.

so implementation wise it would be similar ... still using qualifier (though we could use `@Named` but it is considered less CDI) but instead of linking it with persistence we could just have one that is `@Repository("inmemory")` and another one `@Repository("infinispan")`

and then in application.properties we could then have `kogito.job-service.repository=infinispan` and by default it will be `kogito.job-service.repository=inmemory`

wdyt?

Reply:
and in case we would to this we don't have to specify if the persistence is enabled or nor as it will be taken based on the repository implementation anyway

Reply:
done